components/clarinet-sdk/vitest-helpers/src/formatCV.ts

```diff
@@ -0,0 +1,151 @@
+/*
+  This file could be implement in stacks.js
+  Format stacks.js Clarity Value into Clarity style readable string
+  eg:
+  `Cl.uint(1)` => u1
+  `Cl.list(Cl.uint(1))` => (list u1)
+*/
+
+import {
+  BufferCV,
+  ClarityType,
+  ClarityValue,
+  ListCV,
+  TupleCV,
+  principalToString,
+} from "@stacks/transactions";
+
+let byteToHex: string[] = [];
+for (let n = 0; n <= 0xff; ++n) {
```

Comment:
byteToHex could be imported from stacks/common if needed

Reply:
Ah, I didn't know that, will do.
Also, what do you think about moving formatCV to stacks.js at some point?

Reply:
Let's do it 🏎️ `Cl.toString` 😅

Reply:
I also explored the possibility of having a Cl.toJSON().

Or a custom JSON replacer. And I realize that I actually pushed it but it's dead code -> https://github.com/hirosystems/clarinet/blob/9a4c82bae11fb22c684c7c875678341570970c20/components/clarinet-sdk/vitest-helpers/src/formatCV.ts#L142-L151)

(It has some flaws as it is, and i'll remove it anyway)

So you can do `JSON.stringify(ClarityValue, cvJSONReplacer, 2)`
```
{
  "id": "u1",
  "messageAscii": "\"hello world\"",
  "items": [
    {
      "id": "u1",
      "owner": "(some 'ST2CY5V39NHDPWSXMW9QDT3HC3GD6Q6XX4CFRK9AG)",
    },
  ]
}
```

And it outputs valid JSON while taking care of the indentation (and it would have avoided me to handle the `indent & depth` when formatting lists and tuples.

But what we actually want isn't valid JSON but valid Clarity values (without `""` everywhere and with `(list )` instead of `[]`). Anyway, not sure there is a use for it but interesting to explore.

Reply:
Very interesting!